flips/20201125-bip-44-multi-account.md

```diff
@@ -9,188 +9,279 @@
+should not be used to generate keys for more than one account address.
+
+The Account Index to use when generating a key pair to be added on a new account
+is the smallest Account Index of which has not yet been used to generate keys 
+set on any account.
 
 The motivation for these accounts is the same set forth in BIP 44:
 
-> Users can use these accounts to organize the funds in the same fashion as bank accounts; for donation purposes (where all addresses are considered public), for saving purposes, for common expenses etc.
+> Users can use these accounts to organize the funds in the same fashion as bank
+> accounts; for donation purposes (where all addresses are considered public),
+> for saving purposes, for common expenses etc.
+
+Unlike BIP 44, wallets should not prevent the creation of a new account if any
+existing account corresponding to a previous Account Index does not have any
+transaction history.
 
-In addition to the above, Flow wallets should follow the standards for account numbering, 
-creation and discovery described in [BIP 44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#account).
+In addition to the above, Flow wallets should follow the standards for account
+numbering, creation and discovery described in [BIP
+44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#account).
 
 #### Change
 
-Unlike Bitcoin, Flow does not utilize change addresses. 
-Change is always the constant 0.
+Unlike Bitcoin, Flow does not utilize change addresses. Change is always the
+constant 0.
 
 #### Index
 
-In BIP 44, each index generates a public key that is used to derive
-a unique address for an account.
-Although Flow does not derive account addresses from public keys,
-index can still be used to generate multiple public keys for a single account.
+In BIP 44, each index generates a public key that is used to derive a unique
+address for an account. Although Flow does not derive account addresses from
+public keys, index can still be used to generate multiple public keys for a
+single account.
 
 When applying BIP 44 to Flow, each subsequent value for index should be used to
-generate a new public key for the same account, starting from 0 and incrementing
+generate a new key pair for the same account, starting from 0 and incrementing
 by 1.
 
-This index should not be confused with the on-chain account key index.
+This index should not be confused with the on-chain account key index. It must
+not be assumed that this index will reflect the on-chain account key index for
+any key.
 
-### Account Discovery
+When generating a new key pair for an account, the Key Index to use in the path to 
+generate the new key should be the smallest unused Key Index available which 
+has not yet been used to generate a key on the account.
+
+### Legacy Path
+
+There exist Flow accounts with keys on them which have been generated using the
+Legacy Path.
```

Comment:
I think you can be explicit here and mention that, up until now, the Ledger application created Flow accounts using a different interpretation of BIP 44 that we now refer to as the legacy path. And to our knowledge, this is the only application that created accounts in this way.

Great idea to mention the legacy path here!